.github/actions/setup/action.yml

```diff
@@ -0,0 +1,21 @@
+---
+name: setup
+description: Common tasks for preparing the environment
+
+runs:
+  using: "composite"
+  steps:
+      - uses: actions/setup-go@v5
+        if: runner.os != 'Windows'
+        with:
+          go-version-file: go.mod
+          cache: true
+      # See https://github.com/actions/setup-go/pull/515
+      - uses: antontroshin/setup-go@bda02de8887c9946189f81e7e59512914aeb9ea4
+        if: runner.os == 'Windows'
+        with:
+          go-version-file: go.mod
+          cache: true
```

Comment:
From what I understand, the only thing this fork does is change the location of GOCACHE and GOMODCACHE. Wouldn't it be simpler to use the original action, and just set these variables ourselves?

Reply:
I applied the same pattern we did also for the apm-server at:
- https://github.com/elastic/apm-server/pull/15623

No strong opinion here.

Maybe that could be done in a follow-up?